Adding Cloud Volumes Tech Preview verbiage (#84)
hpe-storage/scod: docs/container_storage_provider/hpe_cloud_volumes/index.md

# Introduction

> The HPE Cloud Volumes CSP integrates seamlessly with the HPE Cloud Volumes Block service in the public cloud. The CSP abstracts the data management capabilities of the storage service for use by Kubernetes. The documentation found herein is mainly geared towards day two operations and reference documentation for the `StorageClass` and `VolumeSnapshotClass` parameters but also contains important HPE Cloud Volumes Block configuration details.

Reviewer: day two - "day-2"

!!! important 
    The HPE Cloud Volumes CSP is currently in **beta** and available as a Tech Preview on Amazon EKS only. Please see the [1.5.0-beta Helm chart](https://artifacthub.io/packages/helm/hpe-storage/hpe-csi-driver/1.5.0-beta).

[TOC]

> !!! seealso

Reviewer: See also

Author: `seealso` is mapped to the blue [admonition](https://github.com/hpe-storage/scod/#admonitions).

    There's a Tech Preview available in the [Video Gallery](../../learn/video_gallery/index.md#tech_preview_hybrid_cloud_kubernetes_using_hpe_cloud_volumes) on how to get started with the HPE Cloud Volumes CSP with the HPE CSI Driver.

## Cloud requirements

> Always check the corresponding CSI driver version in [compatibility and support](../../csi_driver/index.md#compatibility_and_support) for basic requirements (such as supported Kubernetes version and cloud instance OS). If a certain feature is gated against any particular cloud provider it will be called out where applicable.

Reviewer 2: Do we need to specify which version of CSI works with CSP?

Author: This is called out on top of the page. That it's a beta. The CV CSP will be listed as supported on the CSI driver page when we ship 1.5.0.

Consider this table a snapshot of a particular moment in time and consult with the respective hyperscalers and the HPE Cloud Volumes Block service for definitive availability.

!!! note
    In other regions where HPE Cloud Volumes provide services, such as us-west-1, but cloud providers has no managed Kubernetes service; BYO Kubernetes is the only available option when it becomes available as a supported feature of the CSP.

## Limitations

Consult the [compatibility and support](../../csi_driver/index.md#compatibility_and_support) table for generic limitations and requirements. CSI and CSP specific limitations with HPE Cloud Volumes Block is listed below.

- The Volume Group Provisioner and Volume Group Snapshotter sidecars are currently not implemented in the HPE Cloud Volumes CSP.
- The base CSI driver parameter `description` is ignored by the CSP.
- In some cases, your a "regionID" needs to be supplied in the `StorageClass` and in conjunction with Ephemeral Inline Volumes. Your "regionID" may only be found in the APIs. Join us on [Slack](https://slack.hpedev.io/) if you're hitting this issue (it can be seen in the CSP logs).

!!! tip

>     While not a limitation in itself, iSCSI CHAP is mandatory with HPE Cloud Volumes but does not need any CSI driver configuration, the CHAP credentials are queried through the REST APIs from the HPE Cloud Volumes account session and applied automatically during runtime.

Reviewer 2: Is there a need to tell them how we can do it manually by adding it in values.yaml?

Author: I don't think it is. There's no sensible way for a customer to fetch their CHAP details from CV. You either need to make the API call with a HTTP client or "fake" a connection of a CV volumes to a cloud instances and copy out the parameters from cvctl. I rather not elaborate on any of these methods.

>     While not a limitation in itself, iSCSI CHAP is mandatory with HPE Cloud Volumes but does not need any CSI driver configuration, the CHAP credentials are queried through the REST APIs from the HPE Cloud Volumes account session and applied automatically during runtime.

Reviewer: ```suggestion
    While not a limitation, iSCSI CHAP is mandatory with HPE Cloud Volumes but does not need to be configured within the CSI driver. The CHAP credentials are queried through the REST APIs from the HPE Cloud Volumes account session and applied automatically during runtime.
```

Author: Thanks, I'll have this in my next commit.

- [Cloning parameters](#cloning_parameters)
- [Import parameters](#import_parameters)
- [VolumeSnapshotClass parameters](#volumesnapshotclass_parameters)

Please see [using the HPE CSI Driver](../../csi_driver/using.md#base_storageclass_parameters) for base `StorageClass` examples. All parameters enumerated reflects the current version and may contain unannounced features and capabilities.

!!! note
    All parameters are optional unless documented as mandatory for a particular use case.

### Common parameters for provisioning and cloning

These parameters are mutable between a parent volume and creating a clone from a snapshot.

| Parameter                       | String  | Description |
| ------------------------------- | ------- | ----------- |
| destroyOnDelete                 | Boolean | Indicates the backing Cloud Volume (including snapshots) should be destroyed when the PVC is deleted. Defaults to "false" which means volumes needs to be pruned manually in the Cloud Volume service. |
| limitIops                       | Integer | The IOPS limit of the volume. The IOPS limit should be in the range 300 (default) to 20000. |

> | performancePolicy<sup>1</sup>   | Text    | The name of the performance policy to assign to the volume. Available performance policies: "Exchange", "Oracle", "SharePoint", "SQL", "Windows File Server". Defaults to "Other Workloads". |

Reviewer 2: It is not performancePolicy for cv-csp. Instead need to provide retention and schedule

Author: I think you're mixing "performancePolicy" with "protectionSomething"? performancePolicy is part of the parameter list of the CSP.

Reviewer 2: right. I was confused with protectionTemplate

| schedule                        | Text    | Snapshot schedule to assign to the volumes. Available schedules: "hourly", "daily", "twicedaily", "weekly", "monthly", "none". Defaults to "daily". |

> | retentionPolicy                 | Integer | Retention policy to assign to the `schedule`. The parameter must be paired properly with the `schedule`. <br /><br /><ul><li>hourly: 6, 12, 24<li>daily: 3, 7, 14<li>twicedaily: 4, 8, 14<li>weekly: 2, 4, 8<li>monthly: 3, 6, 12</ul>Defaults to "3" paired with the "daily" `retentionPolicy`.

Reviewer 2: Nice

| privateCloud<sup>1</sup>        | Text    | Override the compute instance provided VPC/VNET. |
| existingCloudSubnet<sup>1</sup> | Text    | Override the compute instance provided subnet. |

> | automatedConnection<sup>1</sup> | Boolean | Override the HPE Cloud Volumes configured setting for connection automation. |

Reviewer 2: Do we need to add a note that it gets this value from cloud account if not provided?

Author: I'll add some clarifications.
